app/src/main/java/io/zenandroid/onlinego/ui/screens/settings/CustomAdapter.kt

```diff
@@ -0,0 +1,66 @@
+package io.zenandroid.onlinego.ui.screens.settings
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Build
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.ImageView
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.toArgb
+import io.zenandroid.onlinego.R
+import io.zenandroid.onlinego.data.model.BoardTheme
+
+class CustomAdapter(bordThemes: Array<BoardTheme>, mContext: Context) :
+    ArrayAdapter<BoardTheme?>(
+        mContext, R.layout.row_item_settings_dialog_theme, bordThemes
+    ) {
+    // View lookup cache
+    private class ViewHolder {
+        var themePreview: ImageView? = null
+    }
+
+    @SuppressLint("UseCompatLoadingForDrawables")
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+        // Get the data item for this position
+//        var convertView = convertView
+        val boardTheme = getItem(position)
+        // Check if an existing view is being reused, otherwise inflate the view
+        val viewHolder: ViewHolder // view lookup cache stored in tag
+        val result: View
+        if (convertView == null) {
+            viewHolder = ViewHolder()
+            val inflater = LayoutInflater.from(context)
+            result = inflater.inflate(R.layout.row_item_settings_dialog_theme, parent, false)
+            viewHolder.themePreview =
+                result.findViewById<View>(R.id.theme_preview) as ImageView
+
+            // IMAGE
+            val backgroundImage = boardTheme!!.backgroundImage
```

Comment:
In kotlin we like to avoid double bang (!!) as much as possible as it circumvents the platform's nullability checks. Please try to use ?. or ?.let { someCode() } wherever possible.

Reply:
Ok, never used Kotlin before, I'll look into it to avoid breaking your work